warehouse/manage/views.py

```diff
@@ -279,19 +279,41 @@ def _key(project):
             return project.releases[0].created
         return project.created
 
-    projects_owned = set(
-        project.name
-        for project in (
-            request.db.query(Project.name)
-            .join(Role.project)
-            .filter(Role.role_name == "Owner", Role.user == request.user)
-            .all()
+    projects_owned = (
+        request.db.query(Project)
+        .join(Role.project)
+        .filter(Role.role_name == "Owner", Role.user == request.user)
+        .subquery()
+    )
+
+    with_owner_count = (
+        request.db.query(
+            Role.package_name, func.count(Role.package_name).label("owner_count")
         )
+        .join(projects_owned)
+        .filter(Role.role_name == "Owner")
+        .group_by(Role.package_name)
+        .subquery()
     )
 
+    query_result = (
+        request.db.query(Project.name, with_owner_count.c.owner_count)
+        .join(with_owner_count)
+        .all()
+    )
```

Comment:
I think we could probably add to this query to remove the need for the `for` loop below, using a `groupby` and something like `.having(func.count() == 1)` then calling `.all()` and just setting it to `projects_sole_owned` directly. (Disclaimer, that might not work)

Reply:
The for loop can surely be removed easily but I would have to perform 2 queries (which I'm trying to avoid).
Right now I'm performing a single query to get info for both "projects_owned" and "projects_sole_owned".
Instead of the for loop I could make 2 set comprehensions instead, something like this:
```
projects_owned = set(project for project, _ in query_result)
projects_sole_owned = set(
    project for project, owner_count in query_result if owner_count == 1
)
```
But in my opinion it's not so readable.

From what I've researched, I did not find a way to make your suggestion work :(
Let me know what you this is the best way to proceed.

Reply:
@di just a friendly reminder for this pull request :)

Reply:
Thanks, I will try to investigate this a little more to figure out if the query I'm suggesting is actually possible or not.